src/public/modules/forms/admin/controllers/collaborator-modal.client.controller.js

```diff
@@ -88,20 +91,25 @@ function CollaboratorModalController(
   }
 
   /**
-   * Calls FormAPI update to update the permission list of a form
+   * Calls AdminFormService to update the permission list (collaborators) of a form
    * @param {Array} permissionList - New permission list for the form
    */
   $scope.updatePermissionList = (permissionList) => {
-    return FormApi.update(
-      { formId: $scope.myform._id },
-      { form: { permissionList } },
-    )
-      .$promise.then((savedForm) => {
-        $scope.myform = savedForm
+    return $q
+      .when(
+        AdminFormService.updateCollaborators($scope.myform._id, {
+          permissionList,
+        }),
+      )
+      .then((updatedCollaborators) => {
+        $scope.myform = {
```

Comment:
nit: does just plain assigning the new collaborator list not work?

```js
$scope.myform.permissionList = ...updatedCollaborators
```

but since we are changing the return type to an array we can just do a straight assignment after that change

Reply:
haven't tried; primary reason was cos i heard something about assignment bugging on fe ui and should've cloned so that caused me to clone here HAHAHA